apollo-router/src/plugins/request_decompression.rs

```diff
@@ -0,0 +1,90 @@
+use crate::plugin::{PluginInit, PluginPrivate};
+use crate::services::router;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use tower::{ServiceBuilder, ServiceExt};
+
+#[derive(Debug, Default, Deserialize, JsonSchema, Serialize)]
+struct RequestDecompressionConfig {
```

Comment:
Prefer implementing Default rather than using option. There's history to this where option causes unexpected behaviour in serde for complex objects. There's details in the dev docs (in repo) for config design and implementation.